### DISCUSSION TITLE
File System Route API causes unicode routes to be 404

### DISCUSSION THREAD

**Discussion by dangkyokhoang** [gatsbyjs/gatsby#30439] [Help]
## Description

Gatsby "slugifies" routes by default (https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/#routing-and-linking) which produces unexpected and unwanted routes.

### Steps to reproduce

Create any route using File System Route API, for example: `{Post.slug}.js`.

### Expected result

Accessing routes with unicode characters should work.

### Actual result

Accessing routes with unicode characters returns 404 because the pages that Gatsby creates have "slugified" routes. This should not be the default behavior. If it must be the default behavior, there should be an option to disable it.

### Environment

  System:
    OS: Windows 10 10.0.18363
  Binaries:
    Node: 15.11.0 - C:\Program Files\nodejs\node.EXE
  npmPackages:
    gatsby: 3.1.2 => 3.1.2 
    gatsby-plugin-google-analytics: ^3.1.0 => 3.1.0 
    gatsby-plugin-react-helmet: ^4.1.0 => 4.1.0 
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0 
    gatsby-plugin-remove-trailing-slashes: ^3.1.0 => 3.1.0 
    gatsby-plugin-sass: ^4.1.0 => 4.1.0

**Comment by LekoArts** [comment]
You can override the `gatsby-plugin-page-creator` instance and give it a `slugify` option: https://github.com/gatsbyjs/gatsby/blob/54d4721462b9303fed723fdcb15ac5d72e103778/packages/gatsby-plugin-page-creator/README.md

```js
// gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-page-creator`,
      options: {
        path: `${__dirname}/src/pages`,
        slugify: { /* your options */ },
      },
    },
  ],
}
```

**Reply by dangkyokhoang** [reply]
Thanks for your answer but the plugin doesn't allow disabling `slugify`.

And I got this warning when adding the above config to `gatsby-config.js`:
```
warn Non-deterministic routing danger: Attempting to create page: "/slug", but page "/slug" already exists
This could lead to non-deterministic routing behavior
```

It seems that the above config doesn't override the plugin's default config.

**Reply by LekoArts** [reply]
From our testing it should override the default, you might see this error due to the "remove-trailing-slashes" plugin.

And yes, you can modify the `slugify` options but you can't disable the behavior itself. If you want specific characters to stay the same you could use the custom replacements: https://github.com/sindresorhus/slugify#customreplacements

**Reply by dangkyokhoang** [reply]
May I create a pull request to add disabling feature?
